### ISSUE TITLE
Proposal: Use a dedicated `id` property instead of forcing a unique `name` 

### ISSUE THREAD

**Issue by erezrokah** [cloudquery/plugin-sdk#317]
## Background

To support `overwrite-delete-stale` we've added a constraint for plugins names to be unique.
Also as a result we're planning to make the `path` required, as to allow users to change the `name`. 

## Problem

I believe it's quite [hard to communicate](https://github.com/cloudquery/cloudquery/pull/3276) the reason `name` has to be unique, and easy for users to miss that requirement (also if users split their config across multiple machines we can't really validate it either).

Additionally since we're planning to make `path` required due to the uniqueness of `name`, it adds friction for people onboarding CloudQuery.

## Proposal

Instead of making `name` unique we can add a dedicated `id` configuration (it can default to `name`) and use that instead.
As a result we can achieve the following:

1. Keep the meaning of `name` and `path` the same
2. Keep `path` optional, as we can resolve it from `name`
3. Users that don't need to run CloudQuery in parallel don't need to know anything about uniqueness
4. Easier to understand why an `id` property needs to be unique

TLDR: Only users that split their configuration (e.g. run in parallel) should care about having a unique identifier.

**Comment by erezrokah** [comment]
After an internal discussion we've decided to close this issue and stick with making `path` and `name` unique.
The reason is that are other use cases for needing a field to uniquely identify multiple sources with the same path in a single sync:
1. Distinguish them in the logs
2. Distinguish them in the database

So the meaning of `name` is not actually the plugin name, but the name of the configuration for that source, hence it should be decoupled from path, and path should not infer from name.
